### PR TITLE
os: implement MkdirAll

### PR DESCRIPTION
**Pull request by dkegel-fastly**
Gets "tinygo test path/filepath" closer to passing.

### CODE REVIEW

File: src/os/path_unix.go [hunk 1]
```diff
@@ -34,3 +34,43 @@ func basename(name string) string {
 
 	return name
 }
+
+// splitPath returns the base name and parent directory.
+func splitPath(path string) (string, string) {
```
Comment (dgryski) [inline src/os/path_unix.go]: Seems like splitPath is unused?

**Review by dgryski — APPROVED** [review]
Aside from the splitPath issue, LGTM

**Review by dgryski — APPROVED** [review]
LGTM

**Comment by deadprogram** [comment]
Thanks @dkegel-fastly and @dgryski now merging.

**Pull request merged**